src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -108,4 +111,25 @@ private static CredentialsMatcher githubScanCredentialsMatcher() {
 
     private Connector() {}
 
+    /**
+     * Fail immediately and throw a customized exception.
+     */
+    public static final RateLimitHandler CUSTOMIZED = new RateLimitHandler() {
```

Comment:
@amuniz By the default all connections use `RateLimitHandler.WAIT`([ref](https://github.com/kohsuke/github-api/blob/github-api-1.71/src/main/java/org/kohsuke/github/GitHubBuilder.java#L32)). As part of this PR we have implemented a new RateLimitHandler used until now only for anonymous connection.